pyne/mesh.py

```diff
@@ -1510,6 +1511,102 @@ def _do_op(self, other, tags, op, in_place=True):
 
         return mesh_1
 
+
+class MeshTally(StatMesh):
+    """This class stores all information from all single mesh tally that
+    exists within some meshtal or state point file. Header information is
+    stored as attributes and the "mesh" attribute is a MOAB mesh with all
+    result and relative error data tagged. This class inherits from StatMesh,
+    exposing all statistical mesh manipulation methods.
+    Attributes
+    ----------
+    tally_number : int
+        The tally number.
+        For mesh tally from MCNP, it must end with 4 (e.g. 4, 14, 214).
+        For mesh tally from OpenMC, it could be any int.
+    particle : string
+        Either "neutron" for a neutron mesh tally or "photon" for a photon mesh
+        tally.
+    dose_response : bool
+        True is the tally is modified by a dose response function.
+    x_bounds : list of floats
+        The locations of mesh vertices in the x direction.
+    y_bounds : list of floats
+        The locations of mesh vertices in the y direction.
+    z_bounds : list of floats
+        The locations of mesh vertices in the z direction.
+    dims : list
+        Dimensions of the mesh.
```

Comment:
I'm wondering if some of these types should really be `tuple`s as we generally don't expect them to change.

Reply:
The `x/y/z_bounds` may involve unit conversion. As far as I know, MCNP and OpenMC use `cm` as length unit. But I am not sure whether there is a MC code use `mm` or `m` as unit. And the mesh bounds transformation could also be used to support a moving component/mesh.

But `e_bounds` involves unit conversion, as MCNP uses `MeV` and OpenMC uses `eV`. So, `e_bounds` should be `list` rather than `tuple`.